prov/tcp/src/tcpx_eq.c

```diff
@@ -61,6 +61,28 @@ static struct fi_ops_eq tcpx_eq_ops = {
 	.strerror = ofi_eq_strerror,
 };
 
+int tcpx_eq_match_ep_event(struct slist_entry *entry, const void *arg)
+{
+	struct tcpx_ep *ep = (struct tcpx_ep *) arg;
+	struct util_event *event;
+	fid_t fid;
+
+	event = container_of(entry, struct util_event, entry);
+
+		fid = (event->err)?
+			((struct fi_eq_err_entry *)event->data)->fid:
+			((struct fi_eq_cm_entry *)event->data)->fid;
+
+	return (&ep->util_ep.ep_fid.fid == fid);
+}
+
+void tcpx_eq_remove_ep_events(struct util_eq *eq, struct tcpx_ep *ep)
+{
+	fastlock_acquire(&eq->lock);
+	while(slist_remove_first_match(&eq->list, tcpx_eq_match_ep_event, ep));
```

Comment:
nit: please move ; to line by itself.  It better indicates that this is intentional.

Reply:
realized that I need to free the err_data for fi_eq_err_entries